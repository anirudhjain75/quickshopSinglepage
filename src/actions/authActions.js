import firebase from 'firebase';

export const emailChange = (text) => {
    return {
        type: 'EMAIL_CHANGED',
        payload: text
    };
};

export const passChange = (text) => {
    return {
        type: 'PASSWORD_CHANGED',
        payload: text
    };
};

export const FirstNameChange = (text) => {
    return {
        type: 'FIRST_NAME_CHANGED',
        payload: text
    }
};

export const LastNameChange = (text) => {
    return {
        type: 'LAST_NAME_CHANGED',
        payload: text
    }
};

export const signUp = ({email, password, firstName, lastName}) => {
    return (dispatch) => {
        dispatch({type: "SIGNING_UP"})
        if (email && password) {
            fetch("http://localhost:3000/users", {
                method: "POST",
                headers: {
                    "Accept": "application/json",
                    "Content-Type": "application/json"
                },
                body: JSON.stringify({
                    email,
                    password,
                    firstName,
                    lastName
                })
            })
                .then((response) => console.log(response.json()))
                .then(() => dispatch({
                    type: "LOGIN_SUCCESS"
                }))
        }
    }
};