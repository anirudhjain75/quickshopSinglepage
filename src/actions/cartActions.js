var uuid = require('uuid/v1')

export const addToCart = (item) => {
    return {
        type: 'ADD_TO_CART',
        payload: {...item, uuid: uuid()}
    }
};

export const removeFromCart = (uuid) => {
    return {
        type: "REMOVE_FROM_CART",
        payload: uuid
    }
};