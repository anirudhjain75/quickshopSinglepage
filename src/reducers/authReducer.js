const initialState = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    login: false
};

export default (state = initialState, action) => {
    switch(action.type)
    {
        case 'EMAIL_CHANGED':
            return {
                ...state,
                email: action.payload
            };
        case 'PASSWORD_CHANGED':
            return {
                ...state,
                password: action.payload
            };
        case 'FIRST_NAME_CHANGED':
            return {
                ...state,
                firstName: action.payload
            };
        case 'LAST_NAME_CHANGED':
            return {
                ...state,
                lastName: action.payload
            };
        default:
            return state;
    }
}