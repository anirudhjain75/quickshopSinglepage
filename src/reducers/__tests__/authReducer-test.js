import authReducer from '../authReducer';
import {emailChange, passChange, FirstNameChange, LastNameChange} from "../../actions";

const initialState = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    login: false
};

test("email change", () => {
    const data = "test@test.com";
    expect(authReducer(initialState, emailChange(data))).toMatchSnapshot();
});

test("password change", () => {
    const data = "Minato";
    expect(authReducer(initialState, passChange(data))).toMatchSnapshot();
});

test("first name change", () => {
    const data = "test";
    expect(authReducer(initialState, FirstNameChange(data))).toMatchSnapshot();
});

test("last name change", () => {
    const data = "man";
    expect(authReducer(initialState, LastNameChange(data))).toMatchSnapshot();
});