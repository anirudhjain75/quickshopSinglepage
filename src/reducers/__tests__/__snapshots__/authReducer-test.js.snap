// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email change 1`] = `
Object {
  "email": "test@test.com",
  "firstName": "",
  "lastName": "",
  "login": false,
  "password": "",
}
`;

exports[`first name change 1`] = `
Object {
  "email": "",
  "firstName": "test",
  "lastName": "",
  "login": false,
  "password": "",
}
`;

exports[`last name change 1`] = `
Object {
  "email": "",
  "firstName": "",
  "lastName": "man",
  "login": false,
  "password": "",
}
`;

exports[`password change 1`] = `
Object {
  "email": "",
  "firstName": "",
  "lastName": "",
  "login": false,
  "password": "Minato",
}
`;
